Sources/NIOTransportServices/NIOTSListenerChannel.swift

```diff
@@ -187,11 +185,9 @@ extension NIOTSListenerChannel: Channel {
         }
 
         switch option {
-        case _ as AutoReadOption:
+        case is AutoReadOption:
             return autoRead as! T.OptionType
-        case _ as SocketOption:
-            let optionValue = option as! SocketOption
-
+        case let optionValue as SocketOption:
```

Comment:
These cleanups could be made in the upstream NIO repository as well: it might be nice if you could open a PR to do that.